ccm_web/server/src/api/api.admin.handler.ts

```diff
@@ -105,4 +119,102 @@ export class AdminApiHandler {
     logger.debug(`Time elapsed: (${(end - start) / NS_PER_SEC}) seconds`)
     return finalResult
   }
+
+  async createExternalUser (user: ExternalUserDto, accountID: number): Promise<CanvasUserLoginEmail | APIErrorData | false> {
+    const email = user.email
+    const loginId = email.replace('@', '+')
+    const fullName = `${user.givenName} ${user.surname}`
+    const sortableName = `${user.surname}, ${user.givenName}`
+
+    try {
+      const endpoint = `accounts/${accountID}/users`
+      const method = HttpMethod.Post
+      const body = {
+        user: {
+          // API doesn't provide separate given- and surname fields
+          name: fullName,
+          sortable_name: sortableName,
+          skip_registration: true
+        },
+        pseudonym: {
+          unique_id: loginId,
+          send_confirmation: false
+        },
+        communication_channel: {
+          type: 'email',
+          address: email,
+          skip_confirmation: true
+        },
+        force_validations: false
+      }
+      logger.debug(`Sending admin request to Canvas endpoint: "${endpoint}"; method: "${method}"; body: "${JSON.stringify(body)}"`)
+      const response = await this.requestor.request<CanvasUserLoginEmail>(endpoint, method, body)
+      logger.debug(`Received response with status code (${String(response.statusCode)})`)
+      const {
+        id,
+        name,
+        sortable_name, // eslint-disable-line
+        short_name, // eslint-disable-line
+        login_id // eslint-disable-line
+      } = response.body
+      return { id, name, sortable_name, short_name, login_id, email }
```

Comment:
It looks like you are still "stuffing" `email` to the response from Canvas. I think I'd rather just leave the email out, this is already available in the nested data key. `email` should also be removed from `CanvasUserLoginEmail` (which I'm going to suggest elsewhere can just be merged with `CanvasUser`). `email` doesn't seem to be returned from this request, so it shouldn't be included in the return type passed to`.request`.

Reply:
FWIW, I'm not opposed to modifying or expanding objects from Canvas, but I do want to accurately describe what is returned from endpoints. My approach so far has been to preface any Canvas data shape with `Canvas...` and leave that out with other composites or expansions (e.g `CourseWithSections`).

Reply:
Ha!  I thought I'd removed `email`. Will do again.

Reply:
This was handled in our Zoom work session.

Reply:
🏁